lib/api/funnel.js

```diff
@@ -368,9 +369,29 @@ class Funnel {
    * @returns {Promise<Request>}
    */
   async checkRights (request) {
+   
+    let skipTokenVerification = false;
+    // When the Support of Cookie as Authentication Token is enabled we check if an auth token cookie is present
+    // When a request is made with CookieOnly set to true
+    // We try to use the auth token cookie if present as auth token
+    // otherwise check for auth token as input
+    if ( global.kuzzle.config.security.supportCookieAuthentication
+      && request.input.args.cookieOnly
+      && request.input.headers
+      && request.input.headers.cookie
+    ) {
+      const cookie = Cookie.parse(request.input.headers.cookie);
```

Comment:
`Cookie.parse` may throw if an invalid cookie is provided. This should be caught and a proper `invalid_cookie` error should be returned.